src/qt/guiutil_font.cpp

```diff
@@ -0,0 +1,616 @@
+// Copyright (c) 2014-2025 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/guiutil_font.h>
+
+#include <tinyformat.h>
+#include <util/system.h>
+
+#include <qt/guiutil.h>
+
+#include <QApplication>
+#include <QDebug>
+#include <QFontDatabase>
+#include <QFontMetrics>
+#include <QPointer>
+#include <QWidget>
+
+#include <cmath>
+#include <map>
+#include <memory>
+#include <utility>
+
+namespace GUIUtil {
```

Comment:
it's a bit strange that `fixedPitchFont`, `calculateIdealFontSize`, `TextWidth` are still in guiutil.h; but I guess it's fine.